x/btcstaking/keeper/msg_server.go

```diff
@@ -437,19 +313,134 @@ func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCre
 		return nil, types.ErrInvalidUnbondingTx.Wrapf("unbonding tx fee must be larger that 0")
 	}
 
-	minUnbondingValue := caluculateMinimumUnbondingValue(stakingMsgTx.TxOut[stakingOutputIdx], &vp.Params)
-	if btcutil.Amount(unbondingMsgTx.TxOut[0].Value) < minUnbondingValue {
+	minUnbondingValue := caluculateMinimumUnbondingValue(pm.StakingTx.Transaction.TxOut[stakingOutputIdx], parameters)
+	if btcutil.Amount(pm.UnbondingTx.Transaction.TxOut[0].Value) < minUnbondingValue {
 		return nil, types.ErrInvalidUnbondingTx.Wrapf("unbonding output value must be at least %s, based on staking output", minUnbondingValue)
 	}
 
-	// all good, add BTC undelegation
-	newBTCDel.BtcUndelegation = &types.BTCUndelegation{
-		UnbondingTx:              req.UnbondingTx,
-		SlashingTx:               req.UnbondingSlashingTx,
-		DelegatorSlashingSig:     req.DelegatorUnbondingSlashingSig,
-		DelegatorUnbondingSig:    nil,
-		CovenantSlashingSigs:     nil,
-		CovenantUnbondingSigList: nil,
+	return &ParamsValidationResult{
+		StakingOutputIdx:   stakingOutputIdx,
+		UnbondingOutputIdx: unbondingOutputIdx,
+	}, nil
+}
+
+// CreateBTCDelegation creates a BTC delegation
+func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCreateBTCDelegation) (*types.MsgCreateBTCDelegationResponse, error) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), types.MetricsKeyCreateBTCDelegation)
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// 1. Parse the message into better domain format
+	parsedMsg, err := types.ParseCreateDelegationMessage(req)
+
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+	}
+
+	// 2. Basic stateless checks
+	// - verify proof of possession
+	if err := parsedMsg.ParsedPop.Verify(parsedMsg.StakerAddress, parsedMsg.StakerPK.BbnPk, ms.btcNet); err != nil {
+		return nil, types.ErrInvalidProofOfPossession.Wrapf("error while validating proof of posession: %v", err)
```

Comment:
```suggestion
		return nil, types.ErrInvalidProofOfPossession.Wrapf(err)
```
The message is already in the error text